runtime/src/iree/hal/buffer_heap.c

```diff
@@ -185,6 +185,7 @@ iree_status_t iree_hal_heap_buffer_wrap(
   IREE_ASSERT_ARGUMENT(out_buffer);
   IREE_TRACE_ZONE_BEGIN(z0);
 
+#if !defined(IREE_PLATFORM_EMSCRIPTEN)  // WebGPU buffer data isn't aligned?
```

Comment:
this is a bit too sketchy (and applies to CPU in emscripten, which isn't what we want)

we could add a iree_hal_memory_access_t bit like IREE_HAL_MEMORY_ACCESS_UNALIGNED and disable this check when set (and other checks we may perform in other places)

the calling code wrapping the buffer can then OR in that flag if it knows it's ok to do so

Reply:
Okay, I'll drop that from this PR and keep the patch locally until I implement a better fix (like what you just suggested).

Reply:
Added `IREE_HAL_MEMORY_ACCESS_UNALIGNED`, simple_abs runs again without local patches.